src/fabric/include/fabric.hrl

```diff
@@ -42,3 +42,17 @@
 
 -record(view_row, {key, id, value, doc, worker}).
 -record(change, {key, id, value, deleted=false, doc, worker}).
+
+-type row_property_key() :: id | key | value | doc | worker.
+-type row_properties() :: [{row_property_key(), any()}].
+
+-type view_row_map() ::
+	#{
+	  id => term(),
+	  key => term(),
+	  value => term(),
+	  doc => term(),
+	  worker => term()
```

Comment:
Are any of those mandatory, perhaps those could be `:=`?

Reply:
No, none of these is mandatory.  There is an implicit defaulting mechanism for the missing key-value pairs, which returns `undefined`.